src/k8s-engine/unixlikeIntegrations.ts

```diff
@@ -0,0 +1,141 @@
+import path from 'path';
+import fs from 'fs';
+
+import { Console } from 'console';
+import Logging from '@/utils/logging';
+import resources from '@/resources';
+import PathConflictManager from '@/main/pathConflictManager';
+import * as window from '@/window';
+
+const Integrations = ['helm', 'kim', 'kubectl', 'nerdctl'];
+const console = new Console(Logging.background.stream);
+
+/*
+ * There are probably going to be only two kinds of integrations: WSL for Windows,
+ * and everything else for macos and linux, should it be supported. For now, this class
+ * is currently intended to be used for various VM strategies for macOS.
+ */
+export default class UnixlikeIntegrations {
+  #results: Record<string, boolean | string> = {}
+  /*
+   * Used to supply integration-warnings
+   */
+  protected pathConflictManager = new PathConflictManager();
+
+  /*
+   * We want to watch /usr/local/bin for changes, but fs.watch won't watch it until it exists.
+   * Once there's a watcher on it, the directory can be deleted, and when it's recreated the
+   * original watcher will still work on it.
+   */
+  #binWatcher: fs.FSWatcher|null = null;
+  /*
+   * If there's a successful change in setIntegration, no need to act on the next integration
+   */
+  #ignoreNextChange = false;
+
+  constructor() {
+    this.setupBinWatcher();
+  }
+
+  async listIntegrations(): Promise<Record<string, boolean | string>> {
+    for (const name of Integrations) {
+      const linkPath = path.join('/usr/local/bin', name);
```

Comment:
We use the string `/usr/local/bin` a lot; should we have a constant for that?

Reply:
I'll use a constant in this file, and later on we can use a global constant.